File: 240730/배열의 평균/ave-of-array.c
#include <stdio.h>

int main() {
    // 2행 4열
    int arr[2][4];

    for (int i=0; i<2; i++)
    {
        for (int j=0; j<4; j++)
        {
            scanf("%d ", &arr[i][j]);
        }
    }

    // 가로 평균
    for (int i=0; i<2; i++)
    {
        int cnt = 0, avg = 0;
        for (int j=0; j<4; j++)
        {
            cnt += 1;
            avg += arr[i][j];
        }
        printf("%.1lf ", (double)(avg/cnt));
    }
    printf("\n");
    
    // 세로 평균
    for (int j=0; j<4; j++)
    {
        int cnt = 0, avg = 0;
        for (int i=0; i<2; i++)
        {
            cnt += 1;
            avg += arr[i][j];
        }
        printf("%.1lf ", (double)(avg/cnt));
    }
    printf("\n");

    // 전체 평균
    int cnt = 0, avg=0;
    for (int i=0; i<2; i++)
    {
        for (int j=0; j<4; j++)
        {
            cnt += 1;
            avg += arr[i][j];
        }
    }
    printf("%.1lf ", (double)(avg/cnt));
    
    return 0;
}

File: 240731/배열의 평균/ave-of-array.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        // 초기 배열 짜기
        int[][] arr = new int[2][4];
        for (int i=0; i<2; i++) {
            for (int j=0; j<4; j++) {
                arr[i][j] = sc.nextInt();
            }
        }

        // 가로 평균
        for (int i=0; i<2; i++) {
            double sum = 0;
            for (int j=0; j<4; j++) {
                sum += arr[i][j];
            }
            double avg = sum / 4;
            System.out.printf("%.1f ", avg);
        }
        System.out.print("\n");

        // 세로 평균
        for (int j=0; j<4; j++) {
            double sum = 0;
            for (int i=0; i<2; i++) {
                sum += arr[i][j];
            }
            double avg = sum / 2;
            System.out.printf("%.1f ", avg);
        }
        System.out.print("\n");
        
        // 전체 평균
        double sum = 0;
        for (int i=0; i<2; i++) {
            for (int j=0; j<4; j++) {
                sum += arr[i][j];
            }
        }
        double avg = sum / 8;
        System.out.printf("%.1f",avg);
    }
}